File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
  var answer = Array(targets.length).fill(0);

  targets.forEach((target, index) => {
    let numofTouch = [];

    for (let i = 0; i < target.length; i++) {
      numofTouch = [];  // 초기화
      keymap.forEach(key => {
        
        key.indexOf(target[i]) !== -1
          && numofTouch.push(key.indexOf(target[i]) + 1);
      });
      // console.log("numofTouch", numofTouch, "target: ", target[i]);
      if (numofTouch.length !== 0) {
        answer[index] += Math.min(...numofTouch);
      } else {
        answer[index] = -1;
        break;
      }
      
    }
  });
  console.log(answer);
  return answer;
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
from typing import List

def solution(keymap: List[str], targets: List[str]) -> List[int]:
    key_cost = {}
    for key in keymap:
        for idx, char in enumerate(key):
            key_cost[char] = min(key_cost.get(char, idx + 1), idx + 1)
    result = []
    for target in targets:
        total = 0
        for char in target:
            if char not in key_cost:
                total = -1
                break
            total += key_cost[char]
        result.append(total)
    return result
